Backend/CarRentalApp/CarRentalApp/Services/Registration/RegistrationService.cs

```diff
@@ -14,8 +14,13 @@ public RegistrationService(UserService userService)
             _userService = userService;
         }        
 
-        public async Task<User?> RegisterAsync(UserRegistrationDTO userDTO)
-        {           
+        public async Task<User?> TryRegisterAsync(UserRegistrationDTO userDTO)
+        {
+            if (await _userService.CheckIfExistsAsync(userDTO))
+            {
+                return null;
+            }
+
```

Comment:
Still don't see any value in separate registration service.
Also renaming method doesn't fix issue with the logic: you are using null return value in the controller to return 409 Confilct (User already exists). Is that the only possible outcome during user registration? Now if I want to add any other operations to TryRegisterAsync I can't return null because it expected that null means that user already exists.